src/bases/createWidgetBase.ts

```diff
@@ -117,26 +118,21 @@ const createWidget: WidgetFactory = createStateful
 				const props: VNodeProperties = {};
 
 				this.nodeAttributes.forEach((fn) => {
-					const newProps: VNodeProperties = fn.call(this, assign({}, props));
+					const newProps: VNodeProperties = fn.call(this);
 					if (newProps) {
 						assign(props, newProps);
 					}
 				});
 
-				if (overrides) {
-					assign(props, overrides);
-				}
 				return props;
 			},
 
 			invalidate(this: Widget<WidgetState>): void {
 				const internalState = widgetInternalStateMap.get(this);
-				if (internalState.dirty) {
-					return;
-				}
 				internalState.dirty = true;
 				this.emit({
-					type: 'invalidate'
+					type: 'invalidate',
```

Comment:
Again, thinking out loud... would there ever be a use case that we would want `invalidate` to be cancellable?  I guess if we find a use case for it we can add it.